metabase/metabase
# Add "Replace question" action to dashboard cards

Closes #36497, closes #36586

Adds a "Replace" button to dashboard cards with questions or models. It makes it possible to swap the dashboard card's query while keeping the dashboard layout intact. It's important to call out that we would reset properties of a dash card that no longer makes sense with a new query: parameter mappings, click behavior, visualization settings, and added series.

### How to verify

1. Open any dashboard that has a few cards
2. Enter the editing mode
3. Hover a dashboard card using a question or a model
4. Find and click the "Replace" action in the top-right floating panel
5. Select a new question or a model from the modal picker
6. Ensure the dashboard card's content is replaced, and the query is rerun
7. Ensure parameters auto-wiring worked (your dashboard should have another card using the same table and mapped to some parameters)
8. Ensure you can undo the revert
9. Ensure changes are persisted after clicking "Save"

### Demo

<details>
<summary>Basic</summary>

https://github.com/metabase/metabase/assets/17258145/15363bc3-7a37-49f4-90b6-b3af2096c842

</details>

<details>
<summary>Filters auto-wiring</summary>

https://github.com/metabase/metabase/assets/17258145/38d75e0f-2185-470a-926f-bca7ab74c498

</details>

### Checklist
